src/components/main/Main.js

```diff
@@ -0,0 +1,47 @@
+import "./Main.css";
+
+import Card from "../card/Card";
+import { useState, useEffect } from "react";
+
+function Main() {
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [characters, setCharacters] = useState([]);
+
+  useEffect(() => {
+    fetch("http://hp-api.herokuapp.com/api/characters").then((res) =>
+      res.json().then(
+        (result) => {
+          setIsLoaded(true);
+          setCharacters(result);
+        },
+        (error) => {
+          setIsLoaded(true);
+          setError(error);
+        }
+      )
+    );
```

Comment:
Think it would be better to work with a `.catch`. Not sure if I got all the brackets right though 😄 
```suggestion
    fetch("http://hp-api.herokuapp.com/api/characters").then((res) =>
      res.json()).then(
        (result) => setCharacters(result))
         .catch((error) => {
          setError(error);
        }
      )
    );
```